core/src/main/java/org/commonjava/maven/ext/core/impl/RelocationManipulator.java

```diff
@@ -203,9 +289,298 @@ private void updateDependencyExclusion( ProjectVersionRef depPvr, ProjectVersion
+            if ( result == null )
+            {
+                return Collections.emptyMap();
+            }
+
+            return result;
+        }
+
+        return Collections.emptyMap();
+    }
+
+    /**
+     * Go through the plugin / plugin-execution configurations and find references to the <code>skip</code> parameter for the given Maven plugin
+     * instance.
+     * @return the configurations
+     */
+    private Map<ConfigurationContainer, String> findConfigurations( final Plugin plugin )
+    {
+        if ( plugin == null )
+        {
+            return Collections.emptyMap();
+        }
+
+        final Map<ConfigurationContainer, String> configs = new LinkedHashMap<>();
+        Object configuration = plugin.getConfiguration();
+        if ( configuration != null )
+        {
+            configs.put( plugin, configuration.toString() );
+        }
+
+        final List<PluginExecution> executions = plugin.getExecutions();
+        if ( executions != null )
+        {
+            for ( PluginExecution execution : executions )
+            {
+                configuration = execution.getConfiguration();
+                if ( configuration != null )
+                {
+                    configs.put( execution, configuration.toString() );
+                }
+            }
+        }
+
+        return configs;
+    }
+
+    private Xpp3Dom getConfigXml( final Node node )
+                    throws ManipulationException
+    {
+        final String config = galleyWrapper.toXML( node.getOwnerDocument(), false )
+                                           .trim();
+
+        try
+        {
+            return Xpp3DomBuilder.build( new StringReader( config ) );
+        }
+        catch ( final XmlPullParserException | IOException e )
+        {
+            throw new ManipulationException( "Failed to re-parse plugin configuration into Xpp3Dom: {}. Config was: {}", e.getMessage(), config, e );
+        }
+    }
+
     @Override
     public int getExecutionIndex()
     {
         return 15;
     }
+
+    private final class PluginReference
+    {
+        private final ConfigurationContainer container;
+
+        private final Node groupIdNode;
+
+        private final Node artifactIdNode;
+
+        private final Node versionNode;
+
+        PluginReference( final ConfigurationContainer container, final Node groupIdNode, final Node artifactIdNode,
+                         final Node versionNode )
+        {
+            this.container = container;
+            this.groupIdNode = groupIdNode;
+
+            setGroupId( groupIdNode.getTextContent() );
+
+            this.artifactIdNode = artifactIdNode;
+
+            setArtifactId( artifactIdNode.getTextContent() );
+
+            this.versionNode = versionNode;
+
+            if ( versionNode != null )
+            {
+                setVersion( versionNode.getTextContent() );
+            }
+        }
+
+        public ConfigurationContainer getContainer()
```

Comment:
Don't really need getter/setter for inner class I think?

Reply:
It's using some of these methods on lines 206--210.

Reply:
Getters and setters have been removed.